### PR TITLE
Added default base virtual address

### PR DESCRIPTION
**Pull request by bdevierno1**
Added Default base virtual address

## Summary:
Updated script to set base address to 0x7f000000000. Users may still update the address using the -a flag. Resolves issue 100 and 186

**Usage:**

| This PR includes         |          |
| ------------------------ | -------- |
| Resolves issues          | 👍  NFs crashing when default -a argument not passed in
| Breaking API changes     |
| Internal API changes     |
| Usability improvements   |  
| Bug fixes                |
| New functionality        |
| New NF/onvm_mgr args     | 
| Changes to starting NFs  |  
| Dependency updates       | 
| Web stats updates        | 

## Merging notes:
 - Dependencies: None  

**TODO before merging :**
 - [x] PR is ready for review

## Test Plan:
Ensured that NFs did not crash when running without the user setting address. Ensured that if a default base virtual address value is passed in to the go script, the one that is provided overrides the default. Tested with the diff number of huge pages as well as default ONVM_NUM_HUGEPAGES=1024 as per install guide.

### CODE REVIEW

**Comment by onvm** [comment]
> In response to PR creation

## CI Message

Your results will arrive shortly

File: onvm/go.sh [hunk 1]
```diff
@@ -36,6 +36,11 @@ nf_cores=$3
 SCRIPT=$(readlink -f "$0")
 SCRIPTPATH=$(dirname "$SCRIPT")
 verbosity=1
+#default base virtual address set
+virt_addr="--base-virtaddr=0x7f000000000"
+
```
Comment (dennisafa) [inline onvm/go.sh]: style nit: remove lines 41-43

Comment (bdevierno1) [reply onvm/go.sh]: Thanks Dennis. I'll make the necessary changes.

Comment (dennisafa) [reply onvm/go.sh]: cool. in addition, i think that we also need to tell the user what the default base virt addr is on startup, and that they can override it if necessary.

File: onvm/go.sh [hunk 2]
```diff
@@ -36,6 +36,8 @@ nf_cores=$3
 SCRIPT=$(readlink -f "$0")
 SCRIPTPATH=$(dirname "$SCRIPT")
 verbosity=1
+#default base virtual address set
+virt_addr="--base-virtaddr=0x7f000000000"
```
Comment (dennisafa) [inline onvm/go.sh]: there is an echo output in `function usage {` saying that it adds a virt_addr parameter. we should change the wording to make it clear that there is a base_virt addr default.

File: onvm/go.sh (hunk 1 above)
Comment (bdevierno1) [reply onvm/go.sh]: Ill add this soon.

File: onvm/go.sh [hunk 3]
```diff
@@ -63,6 +65,13 @@ done
 
 verbosity_level="-v $verbosity"
 
+#If Base virtual address has not been set by user. Set to default.
+if [ "${virt_addr}" = "" ]
```
Comment (dennisafa) [inline onvm/go.sh]: Use -z for checking if a bash variable is empty. I.e  if [ -z “virt_addr” ]

**Comment by kevindweb** [comment]
https://serverfault.com/questions/52034/what-is-the-difference-between-double-and-single-square-brackets-in-bash
We should also be using double square brackets as seen in other places in our bash scripts. '[[' not '['. We don't have it everywhere but it is normally safer.

File: onvm/go.sh [hunk 4]
```diff
@@ -63,6 +65,13 @@ done
 
 verbosity_level="-v $verbosity"
 
+#If Base virtual address has not been set by user. Set to default.
```
Comment (kevindweb) [inline onvm/go.sh]:
```suggestion
# If base virtual address has not been set by the user, set to default.
```
Nit again.

File: onvm/go.sh [hunk 5]
```diff
@@ -36,6 +36,8 @@ nf_cores=$3
 SCRIPT=$(readlink -f "$0")
 SCRIPTPATH=$(dirname "$SCRIPT")
 verbosity=1
+#init base virtual address to empty
```
Comment (kevindweb) [inline onvm/go.sh]:
```suggestion
# initialize base virtual address to empty
```
Nit

**Comment by kevindweb** [comment]
Sorry about the nitpicking, but we don't do as much commenting as we should, so we should make it the best it can be.

**Comment by bdevierno1** [comment]
> Sorry about the nitpicking, but we don't do as much commenting as we should, so we should make it the best it can be.

@kevindweb No worries. Helps me learn as well! 👍 Thanks

**Comment by kevindweb** [comment]
@bdevierno1 Thanks for updating, I'll test it in the cluster later this week

**Comment by kevindweb** [comment]
@onvm test it pls

**Comment by onvm** [comment]
> @onvm test it pls

## CI Message

Your results will arrive shortly

**Pull request closed without merging**